### ISSUE TITLE
MetadataClient not supporting SECRET key authorization

### ISSUE THREAD

**Issue by Vumpik** [checkout/checkout-sdk-java#394]
### Environment

* Checkout SDK version: 6.1.0.
* Platform and version: not_relevant
* Operating System and version: not_relevant

### Description

MetadataClientImpl does not support secret key authorization. This is not in line with your API reference documentation https://api-reference.checkout.com/#tag/Card-metadata .

### Expected behavior

MetadataClient would allow usage of secret key authorization same as e.g. PaymentsClientImpl
```  
 public MetadataClientImpl(final ApiClient apiClient, final CheckoutConfiguration configuration) {
        super(apiClient, configuration, SdkAuthorizationType.SECRET_KEY_OR_OAUTH);
    }
```
### Current behavior

``` 
 public MetadataClientImpl(final ApiClient apiClient, final CheckoutConfiguration configuration) {
        super(apiClient, configuration, SdkAuthorizationType.OAUTH);
    }
```

### Possible solution

Change of enum passed into to constructor.

- [ ] I may be able to implement this bug fix

### Additional information

We can go on low level implementation as basic POST request but we would rather stick to the sdk implementation. Or maybe you can suggest any other way to go around this.

**Comment by Vumpik** [comment]
@armando-rodriguez-cko Tried to produce that with low level implementation. Still cannot connect to "metadata/card" with secret key. Getting 401 response code. Tried the same with postman/curl call with the same response. Can you help me with that, or do I need to create new issue?

Same secret key can be use e.g. card registration so it is not in the wrong secret key.

**Comment by armando-rodriguez-cko** [comment]
Yes, of course.

**Comment by armando-rodriguez-cko** [comment]
Your secret key must to have VAULT:CARD-METADATA scope

**Comment by armando-rodriguez-cko** [comment]
Hello @Vumpik, has your problem been solved with the last release 6.1.1?

**Comment by armando-rodriguez-cko** [comment]
We close this issue due inactivity
